dapr_agents/workflow/agentic.py

```diff
@@ -56,44 +54,23 @@ class AgenticWorkflow(
         default=None,
         description="Default topic for broadcasting messages. Set explicitly for multi-agent setups.",
     )
-    state_store_name: str = Field(
-        ..., description="Dapr state store for workflow state."
-    )
-    state_key: str = Field(
-        default="workflow_state",
-        description="Dapr state key for workflow state storage.",
-    )
-    state: dict = Field(
-        default_factory=dict, description="Current state of the workflow."
-    )
-    state_format: Optional[Type[BaseModel]] = Field(
-        default=None,
-        description=(
-            "Optional Pydantic model used to validate the persisted workflow "
-            "state. If provided, state loaded from storage is coerced to this "
-            "schema."
-        ),
-    )
-    agents_registry_store_name: str = Field(
-        ..., description="Dapr state store for agent metadata."
-    )
-    agents_registry_key: str = Field(
-        default="agents_registry", description="Key for agents registry in state store."
+
+    storage: Storage = Field(
```

Comment:
oh so we have storage and memory as separate things. I got confused thinking the `Storage` was related to memory. `Storage` is for state and agent registry right? If so, what if we separate them? StateStorage and RegistryStorage? I would separate them because when I set state and agent registry on the same component, the agent registry gets a prefix (name of the application). When you have multiple agents working together, you have `AgentAPrefix | AgentRegistry`, `AgentBPrefix | AgentRegistry`, `AgentCPrefix | AgentRegistry`. For the state is fine because each state represents the agent, but for a shared agent registry we cannot have a prefix. Using the same component you cannot have both so I keep them separate. Does that make sense?